src/test/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorTest.java

```diff
@@ -204,4 +211,122 @@ void mapMutateKeysFallbackToValues() {
+        defaultTestMapMutator();
+
+    try (MockPseudoRandom prng = mockPseudoRandom()) {
+      Map<Integer, Integer> map = mutator.crossOver(emptyMap(), emptyMap(), prng);
+      assertThat(map).isEmpty();
+    }
+  }
+
+  @Test
+  void testCrossOverInsertChunk() {
+    SerializingMutator<@NotNull Map<@NotNull Integer, @NotNull Integer>> mutator =
+        defaultTestMapMutator();
+
+    Map<Integer, Integer> map = asMap(1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6);
+    Map<Integer, Integer> otherMap = asMap(1, 1, 2, 2, 3, 3, 40, 40, 50, 50, 60, 60);
+
+    try (MockPseudoRandom prng = mockPseudoRandom(
+             // insert action
+             0,
+             // chunk size
+             3,
+             // from chunk offset, will skip first element of chunk as it is already present in map
+             3,
+             // to chunk offset, unused
+             0)) {
+      map = mutator.crossOver(map, otherMap, prng);
+      assertThat(map)
+          .containsExactly(1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 40, 40, 50, 50, 60, 60)
+          .inOrder();
+    }
+  }
+
+  @Test
+  void testCrossOverOverwriteChunk() {
+    SerializingMutator<@NotNull Map<@NotNull Integer, @NotNull Integer>> mutator =
+        defaultTestMapMutator();
+
+    Map<Integer, Integer> map = asMap(1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6);
+    Map<Integer, Integer> otherMap = asMap(1, 1, 2, 2, 3, 3, 40, 40, 50, 50, 60, 60);
+
+    try (MockPseudoRandom prng = mockPseudoRandom(
+             // overwrite action
+             1,
+             // chunk size
+             3,
+             // from chunk offset
+             2,
+             // to chunk offset, will not change first element as values are equal
+             2)) {
+      map = mutator.crossOver(map, otherMap, prng);
+      assertThat(map).containsExactly(1, 1, 2, 2, 3, 3, 4, 40, 5, 50, 6, 6).inOrder();
+    }
+  }
+
+  @Test
+  void testCrossOverCrossOverChunkKeys() {
+    SerializingMutator<@NotNull Map<@NotNull Integer, @NotNull Integer>> mutator =
+        defaultTestMapMutator();
+
+    Map<Integer, Integer> map = asMap(1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6);
+    Map<Integer, Integer> otherMap = asMap(1, 1, 2, 2, 3, 30, 40, 40, 50, 50, 60, 60);
+
+    try (MockPseudoRandom prng = mockPseudoRandom(
+             // cross over action
+             2,
+             // keys
+             true,
+             // chunk size
+             3,
+             // from chunk offset
+             2,
+             // to chunk offset,
+             // first keys are equal and can not be crossed over
+             2,
+             // first key cross over attempts
+             0, 0, 0, 0, 0, 0, 0, 0, 0,
+             // second key cross over
+             0,
+             // third key cross over
+             0)) {
+      map = mutator.crossOver(map, otherMap, prng);
+      assertThat(map).containsExactly(1, 1, 2, 2, 3, 3, 6, 6, 22, 4, 27, 5).inOrder();
+    }
+  }
+
+  @Test
+  void testCrossOverCrossOverChunkValues() {
```

Comment:
I would like to see test coverage for mutable value types as that's a potentially error-prone aspect of this PR.

Reply:
Changed the tests to use mutable lists as keys and values.